packages/next/src/app-dir/links/nextHttp.ts

```diff
@@ -33,11 +33,12 @@ export function experimental_nextHttpLink<
       const revalidate = requestRevalidate ?? opts.revalidate ?? false;
 
       const linkFactory = opts.batch ? httpBatchLink : httpLink;
+      const fetchFn = opts.fetch ?? fetch;
       const link = linkFactory({
         headers: opts.headers as any,
         url: opts.url,
         fetch: (url, fetchOpts) => {
-          return fetch(url, {
+          return fetchFn(url, {
```

Comment:
There's `getFetch` util from `@trpc/client` that you can use here.

Reply:
We need next’s polyfilled fetch in order to use their cache system though right? So I dont see why you would provide a custom fetch function when using this link

Reply:
``` typescript
next: {
  revalidate,
  tags: [cacheTag],
}
```

Maybe take more control of the `next` params.

Or just omit  `fetch` from `opts`.

Reply:
I’d say omit it from types, unless you can provide a concrete example of when you’d want to overwrite the next options (as modifying the cachetag will have to be changed in the revalidate() too which is bound to the client and not the link)

Reply:
Done.